_drafts/arm-cortex-m-exceptions-and-nvic.md

```diff
@@ -0,0 +1,699 @@
+configuration will be bit 1 of the 3<sup>rd</sup> 32-bit register in the bank. Also recall external
+interrupts start at offset 16 in the vector table so the **Exception Number** (index in the vector
+table) for this interrupt will be 16 + 65 = 81.
+
+> Usually, leveraging an NVIC interrupt will be a two step process. You will need to
+> configure the NVIC to enable the interrupt _and_ you will need to configure the MCU specific peripheral to
+> generate the interrupt. You can find more information about that MCU specific configuration side
+> by looking at the specification for the chip you are using. It's also possible to leave the
+> peripheral generation disabled and re-purpose the interrupt to be generated in software using the NVIC registers but this is less common in a real-world application
+
+### Interrupt Set-Enable (NVIC_ISER) and Clear-Enable (NVIC_ICER) Registers
+
+- `NVIC_ISER0`-`NVIC_ISER15`: `0xE000E100`-`0xE000E13C`
+- `NVIC_ICER0`-`NVIC_ICER15`: `0xE000E180`-`0xE000E1BC`
+
+Writing a 1 to the correct bit offset of the register pair will enable or disable the interrupt and a read will return 1 if the interrupt is enabled.
+
+### Interrupt Set-Pending (NVIC_ISPR) and Clear-Pending (NVIC_ICPR) Registers
+
+- `NVIC_ISPR0`-`NVIC_ISPR15`: `0xE000E200`-`0xE000E23C`
+- `NVIC_ICPR0`-`NVIC_ICPR15`: `0xE000E280`-`0xE000E2BC`
+
+Writing a 1 to the correct bit offset of the register pair will set or clear the pending state of the interrupt and a read will return 1 if the interrupt is already pending
+
+### Interrupt Active Bit Registers (NVIC_IABR)
+
+- `NVIC_IABR0`-`NVIC_IABR15`: `0xE000E300`-`0xE000E33C`
+
+A read only bank of registers which return whether or not the interrupt is active. One thing to note is this register is not implemented in the ARMv6-M architecture (Cortex-M0 & Cortex-M0+).
+
+### Interrupt Priority Registers (NVIC_IPR)
+
+- `NVIC_IPR0`-`NVIC_IPR123`: `0xE000E400`-`0xE000E5EC`
+
+The final NVIC configuration register is used to configure the priority of the interrupt. 8 bits
+are used to configure the priority of each interrupt. The number of supported priority levels is
+implementation defined and is in the range of 4-256. When less than 256 priority levels are
+implemented, the lower bits in the field read-as-zero. So, somewhat confusingly, if only 2 bits are implemented, the valid
+values from highest priority to lowest priority would be `0b000.0000` (**0x0**), `0b0100.0000`
+(**0x40**), `0b1000.0000` (**0x80**), `0b1100.0000` (**0xC0**).
+
+## Software Triggered Interrupt Register (STIR) - 0xE000EF00
+
+This register can be used to set an NVIC interrupt to pending. It's equivalent to setting the
+appropriate bit in the `NVIC_ISPR` to 1. The value that needs to be written to the register is the
+**External Interrupt Number** (**Exception Number - 16**). This register is not implemented for the
+ARMv6-M architecture
+
+## Advanced Exception Topics
```

Comment:
To shorten things a little, perhaps worth kicking this section to the start of a follow-on post (though the sub-section about primask basepri faultmask may be useful to keep)... lmk what you think!